src/test/java/riemann/java/client/tests/LocalhostResolveTest.java

```diff
@@ -0,0 +1,146 @@
+    }
+
+    @Test
+    public void testEnvVarWindows() {
+        System.getProperties().put(OS_NAME_PROP, "Windows7");
+
+        String hostname = LocalhostResolver.getResolvedHostname();
+        Assert.assertEquals(ExpectedWinHostname, hostname);
+        Assert.assertEquals(0, LocalhostResolver.getLastNetUpdateTime());
+    }
+
+    @Test
+    public void testEnvVarNix() {
+        System.getProperties().put(OS_NAME_PROP, "Linux");
+        String hostname = LocalhostResolver.getResolvedHostname();
+        Assert.assertEquals(ExpectedNixHostname, hostname);
+        Assert.assertEquals(0, LocalhostResolver.getLastNetUpdateTime());
+    }
+
+    @Test
+    public void testCustomEnvVar() {
+        final String customHostnameEnvVar = "AWS_HOST";
+        final String customHostname = "EC2-LocalHostResolverTest";
+        env.put(customHostnameEnvVar, customHostname);
+        setEnv(env);
+
+        LocalhostResolver.CustomEnvVarName = customHostnameEnvVar;
+        String hostname = LocalhostResolver.getResolvedHostname();
+        Assert.assertEquals(customHostname, hostname);
+        Assert.assertEquals(0, LocalhostResolver.getLastNetUpdateTime());
+    }
+
+
+    /**
+     * evil hack for testing (only!) with env var in-memory modification
+     * see: http://stackoverflow.com/a/7201825/1469004
+     *
+     * @param newEnv - to set in memory
+     */
+    protected static void setEnv(Map<String, String> newEnv) {
+        try {
+            Class<?> processEnvironmentClass = Class.forName("java.lang.ProcessEnvironment");
+            Field theEnvironmentField = processEnvironmentClass.getDeclaredField("theEnvironment");
+            theEnvironmentField.setAccessible(true);
+            Map<String, String> env = (Map<String, String>) theEnvironmentField.get(null);
+            env.putAll(newEnv);
+            Field theCaseInsensitiveEnvironmentField = processEnvironmentClass.getDeclaredField("theCaseInsensitiveEnvironment");
+            theCaseInsensitiveEnvironmentField.setAccessible(true);
+            Map<String, String> cienv = (Map<String, String>)     theCaseInsensitiveEnvironmentField.get(null);
+            cienv.putAll(newEnv);
+        }
+        catch (NoSuchFieldException e) {
+            try {
+                Class[] classes = Collections.class.getDeclaredClasses();
+                Map<String, String> env = System.getenv();
+                for(Class cl : classes) {
+                    if("java.util.Collections$UnmodifiableMap".equals(cl.getName())) {
+                        Field field = cl.getDeclaredField("m");
+                        field.setAccessible(true);
+                        Object obj = field.get(env);
+                        Map<String, String> map = (Map<String, String>) obj;
+                        map.clear();
+                        map.putAll(newEnv);
+                    }
+                }
+            } catch (Exception e2) {
+                e2.printStackTrace();
```

Comment:
but then I have to make every test method declare throw (which I prefer not to here, since it's a hacky test method anyway).
Can I use a RuntimeException?